src/lib/components/video-player.svelte

```
@@ -0,0 +1,97 @@
+<script lang="ts">
+import Hls from "hls.js";
+import { onMount, onDestroy } from "svelte";
+
+// Props
+let { src }: { src: string | null } = $props();
+
+// State
+let videoEl: HTMLVideoElement = $state()!;
+let hls: Hls | null = null; // Not reactive
+let levels: any[] = $state([]);
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)